src/core/models/unlock_block.h

```diff
@@ -10,21 +10,52 @@
 
 // ed25519 signature block  = signature type + public key + signature
 #define ED25519_SIGNATURE_BLOCK_BYTES (1 + ED_PUBLIC_KEY_BYTES + ED_SIGNATURE_BYTES)  // 97 bytes
-// unlock_block_t + reference = 1 + 2
+// unlock_type_t + reference = 1 + 2
 #define UNLOCK_REFERENCE_SERIALIZE_BYTES (1 + sizeof(uint16_t))
-//  unlock_block_t + signature type + pub_key + signature
+//  unlock_type_t + signature type + pub_key + signature
 #define UNLOCK_SIGNATURE_SERIALIZE_BYTES (1 + ED25519_SIGNATURE_BLOCK_BYTES)
+// unlock_type_t + alias index = 1 + 2
+#define UNLOCK_ALIAS_SERIALIZE_BYTES (1 + sizeof(uint16_t))
+// unlock_type_t + NFT index = 1 + 2
+#define UNLOCK_NFT_SERIALIZE_BYTES (1 + sizeof(uint16_t))
 
 /**
- * @brief An unlock block list object
+ * @brief Unlock block types that are supported by the protocol
  *
  */
-typedef struct unlock_blocks {
-  unlock_block_t type;         ///< 0 denotes a Signature Unlock Block, 1 denotes a Reference Unlock Block.
-  uint16_t reference;          ///< Represents the index of a pervious unlock block
-  byte_t* sig_block;           ///< signature type + public key + signature
-  struct unlock_blocks* next;  ///< point to next block
-} unlock_blocks_t;
+typedef enum {
+  UNLOCK_BLOCK_TYPE_SIGNATURE = 0,  ///< Denotes a signature unlock block
+  UNLOCK_BLOCK_TYPE_REFERENCE = 1,  ///< Denotes a reference unlock block
+  UNLOCK_BLOCK_TYPE_ALIAS = 2,      ///< Denotes a alias unlock block
+  UNLOCK_BLOCK_TYPE_NFT = 3         ///< Denotes a NFT unlock block
+} unlock_type_t;
+
+/**
+ * @brief Unlock block referential data (index to unlock block)
+ *
+ */
+typedef struct {
+  uint16_t index;  ///< Represents the index of a previous unlock block or an alias unlock block or a NFT unlock block
+} unlock_referential_data_t;
```

Comment:
don't think this struct is needed, this struct is for distinguishing the opaque `block_data` in `unlock_block_t`, right?
maybe just define a specific type?
```
typedef uint16_t unlock_index_t;
```
that can access the index value via dereferencing `block_data`

Reply:
Good idea. At first this structure has two elements but later one was removed because it was not necessary to have it. Second element was used to set if an unlock block is chainable or not but this can be detected rather with an unlock block type. I have removed the structure.